node/interface.go

```diff
@@ -31,3 +31,10 @@ type P2PAntifloodHandler interface {
 	SetMaxMessagesForTopic(topic string, maxNum uint32)
 	IsInterfaceNil() bool
 }
+
+// Accumulator defines the interface able to accumulate data and periodically evict them
+type Accumulator interface {
+	AddData(data interface{})
+	OutputChan() <-chan []interface{}
```

Comment:
do you really need access to the channels ? and why interface{} everywhere ? TransactionHandler ?

Reply:
@bogdan-rosianu nailed it. It is supposed to be used in other use-cases also, that is why the implementation is placed in core package. I need to access the chan as to be able to have a stopping mechanism & an easy to write iterator